### DISCUSSION TITLE
Forward SOAP request using YARP.

### DISCUSSION THREAD

**Discussion by nramkrishna** [dotnet/yarp#2216] [Q&A] [answered]
Appreciate if some one can help me with this. I'm trying to forward a request to SOAP service web method and can't find any reference on how to do this. The SOAP service has username and password headers included.  SOAP request looks like below 
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" other namespaces">
   <soapenv:Header>
      <a:Authentication>
         <a:Username>?</a:Username>
         <a:Password>?</a:Password>
      </a:Authentication>
   </soapenv:Header>
   <soapenv:Body>
      <a:test>
         <a:id>?</a:id>
         
         <a:Info>?</a:Info>
      </a:test>
   </soapenv:Body>
</soapenv:Envelope>

Service Url (example): "https://wwww.sample.asmx"
WebMethhod: "Test"

Need help in how to configure so the request can send to soap service test method.
Thanks for the help in advance.

**Comment by Tratcher** [comment]
SOAP -> YARP -> SOAP? That should just work, YARP only looks at the outer HTTP. Or do you need to re-write the request body / SOAP message to add these fields? That's harder.

https://microsoft.github.io/reverse-proxy/articles/transforms.html#request-body-transforms

**Reply by nramkrishna** [reply]
Thank you for your response!
I have to rewrite the soap message.  Any thoughts on how to accomplish that?
Is it straight forward to send the SOAP request through proxy to SOAP service (SOAP -> YARP -> SOAP)? Please can you add sample/example code or config for this scenario?

**Reply by nramkrishna** [reply]
Appreciate any help for above scenario.

**Reply by nramkrishna** [reply]
Appreciate if anyone give me a sample code/config for SOAP -> YARP -> SOAP scenario. Thanks, in advance.

**Reply by Tratcher** [reply, marked as answer]
SOAP is sent as HTTP to YARP, it doesn't require any special handling. To modify the body of the request (the SOAP message) see:

https://microsoft.github.io/reverse-proxy/articles/transforms.html#request-body-transforms shows how to buffer the request body and replace it. To modify it in the middle you could use an XML parser.

**Reply by nramkrishna** [reply]
Thank you! Appreciate your help!
